src/scenarios/billing/SubscribeToSubscriptionPlan/SubscribeToSubscriptionPlan.tsx

```diff
@@ -0,0 +1,476 @@
+
+  const apolloClient = getApolloClient(
+    new URL('graphql', activeProfile.billingServiceBaseURL).href,
+  );
+
+  useEffect(() => {
+    const pathname = window.location.pathname;
+    const urlParams = new URLSearchParams(window.location.search);
+    const subscriptionId = urlParams.get('subscriptionId');
+
+    if (subscriptionId) {
+      logger.log('Redirection from Paypal detected.');
+      if (pathname.endsWith('subscribe-success')) {
+        logger.log(
+          `Subscription with ID ${subscriptionId} should be active now. You can verify it by using 'List Subscription Plans' scenario.`,
+        );
+      } else if (pathname.endsWith('subscribe-error')) {
+        logger.log(
+          `An error occurred in subscription with ID ${subscriptionId}.`,
+        );
+      }
+    }
+  }, [logger]);
+
+  const fetchPaymentPlans = async (): Promise<void> => {
+    try {
+      const result = await apolloClient.query({
+        query: getPaymentPlansQuery,
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      logger.log('calling [fetchPaymentPlans]', 'output:', result.data);
+      if (result.errors) {
+        logger.error('calling [fetchPaymentPlans]', 'output:', result.errors);
+      } else {
+        setPaymentPlans(result.data.paymentPlans.nodes);
+
+        const paypalSettings = result.data.paypalSettings.nodes;
+        if (Array.isArray(paypalSettings) && paypalSettings.length > 0) {
+          setPaypalClientId(paypalSettings[0].clientId);
+        }
+      }
+    } catch (error) {
+      setPaymentPlans([]);
+      if (error instanceof Error) {
+        logger.error('calling [fetchPaymentPlans]', 'output:', error.message);
+      }
+    }
+  };
+
+  const fetchPaymentPlanPrices = async (
+    paymentPlanId: string,
+  ): Promise<void> => {
+    try {
+      const result = await apolloClient.query({
+        query: getPaymentPlanPricesQuery,
+        variables: {
+          id: paymentPlanId,
+        },
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      logger.log('calling [fetchPaymentPlanPrices]', 'output:', result.data);
+      if (result.errors) {
+        logger.error(
+          'calling [fetchPaymentPlanPrices]',
+          'output:',
+          result.errors,
+        );
+      } else {
+        setPaymentPlanPrices(result.data.paymentPlan.prices.nodes);
+      }
+    } catch (error) {
+      setPaymentPlanPrices([]);
+      if (error instanceof Error) {
+        logger.error(
+          'calling [fetchPaymentPlanPrices]',
+          'output:',
+          error.message,
+        );
+      }
+    }
+  };
+
+  const changePurchaseFlow = async (
+    newPurchaseFlow: PurchaseFlow,
+  ): Promise<void> => {
+    if (newPurchaseFlow === 'POPUP') {
```

Comment:
we can consider refactoring out the popup-flow related methods into a util file since the scenario file becomes bit too long now